flasc/energy_ratio/energy_ratio.py

```diff
@@ -655,11 +666,33 @@ def _get_energy_ratio_single_wd_bin_bootstrapping(
     if N <= 1:
         results_array = np.array([energy_ratio_nominal] * 3, dtype=float)
     else:
+
+        # If using block-bootstrapping, set up blocks
+        if block_bootstrapping:
+            block_list = list(range(num_blocks))  # List of all block names
+            block_indices = np.arange(df_binned.shape[0]) # Simple index for iloc
+            block_length = int(len(block_indices) / num_blocks) # Length of each block
+            block_labels = np.zeros(len(block_indices)).astype(int) # Labels to assign each index
+            for b_i in range(num_blocks):
+                block_labels[block_length * b_i:block_length * (b_i+1)] = b_i
+
         # Get a bootstrap sample of range
         bootstrap_results = np.zeros(N)
         bootstrap_results[0] = energy_ratio_nominal
         for i in range(1, N):
-            df_randomized = df_binned.sample(frac=1, replace=True).copy()
+            if not block_bootstrapping:
+                df_randomized = df_binned.sample(frac=1, replace=True).copy()
+            else:
+                test_blocks = choices(block_list, k = num_blocks) # Choose a set of blocks
```

Comment:
I like this and I think this can also work for regular bootstrapping if num_blocks == num_samples.